packages/core/src/MachineNode.ts

```diff
@@ -187,13 +169,13 @@ export class MachineNode<
    *
    * @param state The state to resolve
    */
-  public resolveState(state: State<TContext, TEvent>): State<TContext, TEvent> {
+  public resolveState(state: State<TContext, TEvent, any>): typeof state {
```

Comment:
using `typeof` to type the return value - nice 👍 I don't think I've ever used it like this and it's quite nice